src/coverage.ts

```diff
@@ -0,0 +1,81 @@
+import { once } from 'events';
+import { fork, execSync } from 'child_process';
+import path from 'path';
+import { existsSync, renameSync } from 'fs';
+import { removeSync, moveSync, copySync } from 'fs-extra';
+import exitHook from 'exit-hook';
+
+export function cleanUp(): void {
+  if (existsSync('./contracts-backup/')) {
+    moveSync('./contracts-backup', './contracts', { overwrite: true });
+  }
+  removeSync('./contracts-backup/');
+  removeSync('./build/contracts/');
+  removeSync('./.coverage_artifacts/');
+  removeSync('./.coverage_contracts/');
+}
+
+exitHook(cleanUp);
+
+/* eslint-disable @typescript-eslint/no-var-requires */
+export async function runCoverage(skipFiles: string[], compileCommand: string, testCommand: string[]): Promise<void> {
+  const client = require('ganache-cli');
+  const CoverageAPI = require('solidity-coverage/api');
+  const utils = require('solidity-coverage/utils');
```

Comment:
Any particular reason why these are done inside the function rather than outside as `import` statements? I think `require` returns `any`.

Reply:
The reason is to avoid a dependency on `ganache-cli` and `solidity-coverage`, if you are not using coverage in your project.